tracer/src/Datadog.Trace/Ci/GitCommandHelper.cs

```diff
@@ -424,6 +424,204 @@ bool IsDefaultBranch(string candidate) => !StringUtil.IsNullOrWhiteSpace(default
+            {
+                commitMessage = commitMessage.Substring(0, commitMessage.Length - 1).Trim();
+            }
+
+            var commitData = new CommitData(
+                CommitSha: commit,
+                AuthorDate: DateTimeOffset.FromUnixTimeSeconds(authorUnixDate),
+                AuthorName: gitLogDataArray[2],
+                AuthorEmail: gitLogDataArray[3],
+                CommitterDate: DateTimeOffset.FromUnixTimeSeconds(committerUnixDate),
+                CommitterName: gitLogDataArray[5],
+                CommitterEmail: gitLogDataArray[6],
+                CommitMessage: commitMessage);
+
+            Log.Debug("GitCommandHelper.FetchCommitData: completed successfully for {Commit}", commitSha);
+            return commitData;
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "GitCommandHelper.FetchCommitData: unexpected error while fetching data for {Commit}", commitSha);
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Determines whether the repository located at <paramref name="workingDirectory"/> is a shallow clone.
+    /// </summary>
+    /// <param name="workingDirectory">Path to the git working directory.</param>
+    /// <returns>True if the repository is a shallow clone; otherwise, false.</returns>
+    public static bool IsShallowCloneRepository(string workingDirectory)
+    {
+        // We need to check if the git clone is a shallow one before uploading anything.
+        // In the case is a shallow clone we need to reconfigure it to upload the git tree
+        // without blobs so no content will be downloaded.
+        var gitRevParseShallowOutput = RunGitCommand(
+            workingDirectory,
+            "rev-parse --is-shallow-repository",
+            MetricTags.CIVisibilityCommands.CheckShallow);
+        if (gitRevParseShallowOutput is null)
+        {
+            Log.Warning("GitCommandHelper: 'git rev-parse --is-shallow-repository' command is null");
+            return false;
+        }
+
+        return gitRevParseShallowOutput.Output.IndexOf("true", StringComparison.OrdinalIgnoreCase) > -1;
+    }
+
+    /// <summary>
+    /// Parses and returns the git version (major, minor, patch) as reported by <c>git --version</c>.
+    /// </summary>
+    /// <param name="workingDirectory">Path to the git working directory.</param>
+    /// <returns>VersionInfo containing the major, minor, and patch version numbers.</returns>
+    public static VersionInfo GetGitVersion(string workingDirectory)
+    {
+        var output = RunGitCommand(
+            workingDirectory,
+            "--version",
+            MetricTags.CIVisibilityCommands.GetBranch);
+
+        if (output is { ExitCode: 0 } && !string.IsNullOrWhiteSpace(output.Output))
+        {
+            // Expected format: "git version 2.41.0" or similar
+            var parts = output.Output.Trim().Split(' ');
+            var versionText = parts.Last();
```

Comment:
Just dropping this here...

```csharp
var span = output.Output.AsSpan().Trim();
var lastSpace = span.LastIndexOf(' ');
var versionText = span.Slice(lastSpace + 1).ToString();
```

Also, _totally_ fine not to optimise any of this seeing as it's called once, the `Split`s just hurt my soul 😛

Reply:
done in [c5102c4](https://github.com/DataDog/dd-trace-dotnet/pull/7285/commits/c5102c40f83d21ccbbcabc2d34f362b2045481b0)